### PR TITLE
Prompts Picker

### PR DESCRIPTION
**Pull request by thenamankumar**
closes: https://linear.app/sourcegraph/issue/SRCH-1264/surface-prompts-in-cmd-k-in-vscode-sorted-by-last-used

Adds quick pick for prompts with `alt+p` shortcut.

https://www.loom.com/share/18c0c54259e34704834d47544e02b2a3?sid=b112f153-48f2-4d4c-9a1c-0e248252ccf0

## Test plan

- alt + p
- search and select any prompt
- it should be executed the same way it does from the prompt library. 

## Changelog

Adds quick pick for prompts with `alt+p` shortcut.

### CODE REVIEW

**Comment by taiyab** [comment]
@thenamankumar Added a few comments. I'd like us to try and get them in as they vastly improve UX for little cost.

**Comment by thenamankumar** [comment]
> @thenamankumar Added a few comments. I'd like us to try and get them in as they vastly improve UX for little cost.

- I updated the short cut to `Shift+Opt+P`.
- I also moved the description to below the title.
- Lets add the additional links like "New Prompt" later separately. I don't have more time this week. 
CleanShot 2024-11-20 at 16 53 25@2x

**Comment by taiyab** [comment]
> > @thenamankumar Added a few comments. I'd like us to try and get them in as they vastly improve UX for little cost.
> 
> * I updated the short cut to `Shift+Opt+P`.
> * I also moved the description to below the title.
> * Lets add the additional links like "New Prompt" later separately. I don't have more time this week.
>   CleanShot 2024-11-20 at 16 53 25@2x

Aweesome! No worries.

**Review by vovakulikov — APPROVED** [review]

**Comment by kalanchan** [comment]
Thanks all! I'm going to merge this as its a required feature we need to ship on Nov 25

File: vscode/src/chat/chat-view/ChatsController.ts [hunk 1]
```diff
@@ -102,6 +103,26 @@ export class ChatsController implements vscode.Disposable {
...
+        text,
+        mode,
+        autoSubmit,
+    }: { text: string; mode: PromptMode; autoSubmit: boolean }): Promise<void> {
+        await vscode.commands.executeCommand('cody.chat.new')
+
+        const webviewPanelOrView =
+            this.panel.webviewPanelOrView || (await this.panel.createWebviewViewOrPanel())
+
+        setTimeout(
```
[the remark this answers is not in the document]
Comment (thenamankumar) [reply vscode/src/chat/chat-view/ChatsController.ts:116]: If there is no Cody panel opened, we need to create a new panel and render the webview. Thats way a delay in setting the editor state and more.

**Pull request merged**